pkg/skr/gcpnfsvolumerestore/findRestoreOperation.go

```diff
@@ -0,0 +1,66 @@
+package gcpnfsvolumerestore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/kyma-project/cloud-manager/pkg/util"
+	"google.golang.org/api/googleapi"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kyma-project/cloud-manager/api/cloud-resources/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+)
+
+func findRestoreOperation(ctx context.Context, st composed.State) (error, context.Context) {
+	state := st.(*State)
+	logger := composed.LoggerFromCtx(ctx)
+
+	//If deleting, continue with next steps.
+	if composed.MarkedForDeletionPredicate(ctx, st) {
+		return nil, nil
+	}
+
+	restore := state.ObjAsGcpNfsVolumeRestore()
+	opName := restore.Status.OpIdentifier
+	logger.WithValues("Nfs Restore source:", restore.Spec.Source.Backup.ToNamespacedName(state.Obj().GetNamespace()),
+		"destination:", restore.Spec.Destination.Volume.ToNamespacedName(state.Obj().GetNamespace())).Info("Finding GCP Restore Operations")
+
+	//If no OpIdentifier, then continue to next action.
+	if opName != "" {
+		return nil, nil
+	}
+
+	project := state.Scope.Spec.Scope.Gcp.Project
+	dstLocation := state.GcpNfsVolume.Status.Location
+	nfsInstanceName := fmt.Sprintf("cm-%.60s", state.GcpNfsVolume.Status.Id)
+	op, err := state.fileRestoreClient.FindRestoreOperation(ctx, project, dstLocation, nfsInstanceName)
+	if err != nil {
+		var e *googleapi.Error
+		if ok := errors.As(err, &e); ok {
+			if e.Code == 404 {
+				return nil, nil
+			}
+		}
```

Comment:
Please consider using this method: https://github.com/kyma-project/cloud-manager/blob/de90d6cfd2d3a107728b131769375414245d2d47/pkg/kcp/provider/gcp/meta/metadata.go#L9

Reply:
Done.